tests/python-e2e/ops.py

```diff
@@ -18,6 +18,49 @@ def test_read_write_family(self):
             self.assertEqual(read_text, TEXT)
             rw_file.close()
 
+    def test_fopen(self):
+        """
+        Opens a file in "/tmp/"
+        """
+        file_path, _ = self._create_new_tmp_file()
+        self.assertFalse(self._check_path_exists_on_host(file_path))
+        with open(file_path, "r") as f:
+            f.seek(0)
+            txt = f.readline()
+        self.assertEqual(txt, TEXT)
+
+    def test_fdopen(self):
+        """
+        Opens a file in "/tmp"
+        """
+        file_path, _ = self._create_new_tmp_file()
+        self.assertFalse(self._check_path_exists_on_host(file_path))
+        fd = os.open(file_path, os.O_RDONLY)
+        self.assertEqual(fd, 3)
+        fd.close()
+
+    def test_fread(self):
+        """
+        Reads a file to test `fread()`
+        """
+        file_path, _ = self._create_new_tmp_file()
+        self.assertFalse(self._check_path_exists_on_host(file_path))
+        fd = os.open(file_path, os.O_RDONLY)
+        self.assertEqual(fd, 3)
+        fd.close()
+
+    def test_fileno(self):
+        """
+        Gets a fd from a file stream
+        """
+        file_path, _ = self._create_new_tmp_file()
+        self.assertFalse(self._check_path_exists_on_host(file_path))
+        fd = os.open(file_path, os.O_RDONLY)
+        self.assertEqual(fd, 3)
+        with open(file_path, "r") as f:
+            self.assertEqual(fd, f.fileno())
```

Comment:
This is weird, does the fake fd exist in the process's open file descriptors? iirc when i was testing this, i didnt see it.
can you print `os.getpid()` and the `fd` after opening the file and add a `sleep` then paste the output of /proc/pid/fd here?

Reply:
I can't seem to get the E2E tests to run in the workflow to print it out. I'll keep working on that.